### PR TITLE
improve performance for search with many intervals

### PR DESCRIPTION
**Pull request by hanars**
Changed a Mendiliome search that failed locally after 500 seconds to finishing in 90 seconds 🤷‍♀️

### CODE REVIEW

File: hail_search/queries/mito.py [hunk 1]
```diff
@@ -197,6 +198,8 @@ def _parse_variant_keys(self, variant_ids=None, **kwargs):
     def _prefilter_entries_table(self, ht, parsed_intervals=None, exclude_intervals=False, **kwargs):
         if exclude_intervals and parsed_intervals:
             ht = hl.filter_intervals(ht, parsed_intervals, keep=False)
+        elif len(parsed_intervals or []) >= MAX_LOAD_INTERVALS:
```
Comment (bpblanken) [inline hail_search/queries/mito.py:201]: do we know why this is faster?

[the docs](https://hail.is/docs/0.2/guides/genetics.html#using-hl-filter-intervals) say this is `suitable for a small list of intervals`, and we're doing the opposite.

Comment (hanars) [reply hail_search/queries/mito.py:201]: No idea, its very counterintuitive. I asked on zulip - https://hail.zulipchat.com/#narrow/stream/123010-Hail-Query-0.2E2-support/search/_intervals.20filter_intervals

File: hail_search/queries/mito.py [hunk 2]
```diff
@@ -135,7 +136,7 @@ def __init__(self, *args, **kwargs):
 
     def _parse_intervals(self, intervals, exclude_intervals=False, **kwargs):
         parsed_intervals = super()._parse_intervals(intervals,**kwargs)
-        if parsed_intervals and not exclude_intervals:
+        if parsed_intervals and not exclude_intervals and len(parsed_intervals) < MAX_LOAD_INTERVALS:
```
Comment (bpblanken) [inline hail_search/queries/mito.py:139]: is `load_table_kwargs` only passed to the annotations read table and not the entries read table?

Comment (hanars) [reply hail_search/queries/mito.py:139]: no its *only* passed to the entries read table. the annotations table is not read in, its just accessed via query_table

**Review by bpblanken — APPROVED** [review]

**Pull request merged**